api/locks.go

```diff
@@ -66,10 +71,22 @@ type Locks struct {
 	c        *Client
 	variable Variable
 	ttl      time.Duration
+	ro       retryOptions
 
 	WriteOptions
 }
 
+type LocksOption = func(l *Locks)
+
+// WithMaxRetries allows access to configure the number of max retries the lock
+// handler will perform in case of an expected response while interacting with the
+// locks endpoint.
+func WithMaxRetries(maxRetries int64) LocksOption {
```

Comment:
Because this is a free function and not a method, the name `WithMaxRetries` is going to be in the package scope and not scoped to the stuff in this file. I could see us wanting to have something like this for other options structs in the future, which would make this generic name awkward to workaround. I'd recommend we either name this so that it's scope is obvious, or somehow make it a method on `Locks`

Reply:
Makes a lot of sense, let me figure it out :)